File: Andy/src/level1/java/Pokemon.java
package level1.java;

import java.util.HashSet;
import java.util.Set;

//프로그래머스 Level 1,폰켓몬
public class Pokemon {
    public int solution(int[] nums) {
        int answer = 0;      
        int cnt = nums.length/2;
        Set<Integer> set = new HashSet<>();
        for(int d:nums) set.add(d);
        if(cnt<set.size()) answer = cnt;
        else answer = set.size();
        return answer;
    }
}

File: Andy/src/level1/javascript/Pokemon.js
// 프로그래머스 Level 1,폰켓몬
function solution(nums) {
    var answer = 0;
    var cnt = parseInt(nums.length/2);
    var set = new Set(nums);
    if(cnt<set.size) answer=cnt;
    else answer = set.size;
    return answer;
}